packages/vuelidate/src/core.js

```diff
@@ -106,31 +107,31 @@ function createComputedResult (rule, model) {
  * @param {Ref<Boolean>} $pending
  * @return {Ref<Boolean>}
  */
-function createAsyncResult (rule, model, initResult, $pending) {
-  const $invalid = ref(true)
+function createAsyncResult (rule, model, $pending, $dirty) {
+  const $invalid = ref(!!$dirty.value)
+  const $pendingCounter = ref(0)
```

Comment:
Just wondering what happens, if you drop the counter? Is it to deal with racing conditions?

Reply:
It’s to prevent setting `$pending` to `false`, when there were multiple requests done and one already returned (and for example was `$invalid === false`), thus giving a false positive, since the later requests could return `$invalid === true`. So what happens here is just making sure that `$pending` is always `true` where there is at least 1 request still going on.

Reply:
If we could cancel the previous request before firing the new one, that would be pretty dope :/ This is something that RX does pretty nicely too.

How would a userland cancellation look like?

```js
const cancelToken = axio.createCancelToken()

async asyncValidator() {
  cancelToken.cancel()
 try {
     const response =  await axios('place', { cancel: cancelToken.token })
  } catch(e) {
     // could also handle the cancel token or smthn
     return false
  }
  return response.isValid
}
```

Reply:
Let’s introduce that in another PR to unblock this one.